### PR TITLE
Add Chat REST API

### PR DESCRIPTION
**Pull request by deff7**

### CODE REVIEW

File: core/block/chats/chatsubscription/manager.go [hunk 1]
```diff
@@ -292,6 +303,19 @@ func (s *subscriptionManager) Flush(reloadStateIfNeeded bool) {
...
+		sseEvents := cloneEvents(events)
+		eventsSetSubIds([]string{ss.id}, sseEvents)
+		ev := &pb.Event{
+			ContextId: s.chatId,
+			Messages:  sseEvents,
+		}
+		select {
+		case ss.sink <- ev:
+		default:
+		}
```
[the remark this answers is not in the document]
Comment (requilence) [reply core/block/chats/chatsubscription/manager.go:325]: @deff7 i think it may be good to replace default with timeout here, and explicitly close the sse connection after

File: core/api/service/chat.go [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+	return messages, nil
+}
+
+// EnrichChatMessageCreators rewrites each message's Creator field from the raw
+// identity emitted by the middleware to the deterministic participant id, and
+// fills CreatorName by batch-querying participant objects in the space.
+// Messages with no creator are left untouched. Identities that do not resolve
+// to a participant object still receive a participant id (it is computable
+// without an RPC) but an empty CreatorName.
+func (s *Service) EnrichChatMessageCreators(ctx context.Context, spaceId string, msgs []apimodel.ChatMessage) {
```
Comment (requilence) [inline core/api/service/chat.go:111]: lets have a single participant subscription. we already have it for types and  properties in the api service

**Review by requilence — CHANGES_REQUESTED** [review]

**Comment by requilence** [comment]
Integration tests for this PR are up in **anytype-suite#41**: https://github.com/anyproto/anytype-suite/pull/41 — 15 test files covering all chat endpoints + 7 SSE edge-case scenarios. Currently **127 / 133 passing** against this branch; the remaining 6 failures are documented as suspected server-side issues (annotated inline with `// SERVER BUG SUSPECTED:` blocks pointing to the exact handler line). Summary of what needs to change on this branch:

## 1. OpenAPI ↔ handler ↔ middleware disagree on chat `limit`  *(4 failing tests)*

Three sources of truth disagree on the `limit` query parameter for `/messages` and `/messages/stream`:

- **`core/api/docs/openapi.json`**: declares `limit` as `default: 50, type: integer` with **no `minimum`, no `maximum`** — clients reading the spec see "any integer accepted".
- **`core/api/handler/chat.go::GetChatMessagesHandler` + `chat_stream.go::ChatStreamHandler`**: explicit `if parsed > 0 && parsed <= 1000` branch that silently falls back to default 50 on out-of-range / unparsable values.
- **Live behavior**: `core/api/server/router.go`'s `v1.Use(paginator)` intercepts first and returns **400** on `limit < 1` or `limit > 1000`. The handler's silent-fallback branch is unreachable dead code.

**Suggested fix** — pick one model and align all three layers:
1. Add `minimum: 1, maximum: 1000` to the `limit` schema in `openapi.json` for `/messages` and `/messages/stream`.
2. Keep the middleware's 400 as the documented contract.
3. Delete the dead silent-fallback branches in `GetChatMessagesHandler` and `ChatStreamHandler`.

## 2. `?before_order_id={X}&after_order_id={X}` ignores the after-bound  *(1 failing test)*

With both cursors set to the same value, the expected open-open interval `(X, X)` is empty. Server returns 5 messages (those with `order_id < X`), ignoring `after_order_id`.

**Suggested fix**: apply both cursor bounds when both are set.

## 3. Mark range validation returns 500 instead of 400  *(2 failing tests)*

Server correctly validates and rejects inverted (`from > to`) and out-of-bounds mark ranges in `POST /messages` — but the response is `500 internal_server_error`. This is client-input validation, not a server failure; 4xx is the correct class.

**Suggested fix**: map mark-range validation errors to 400 in `AddChatMessageHandler` and the matching `EditChatMessageHandler` PATCH path via the existing `util.MapErrorCode` pattern (add e.g. `ErrInvalidMarkRange` to `core/api/service/chat.go`).

## Chat creation

Right now it's only possible to create chat this way
```
POST /v1/spaces/{spaceId}/objects   { "type_key": "chat_derived", "name": "..." }   → chatId
```

I suggest to add  `POST /v1/spaces/{space_id}/chats` to create chat to make things easier (chat_derived is kinda ugly, it was a mistake that we left it for normal non-derived chats)

**Review by requilence — APPROVED** [review]
